rafthttp/linkstatus.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rafthttp
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+
+	"github.com/coreos/etcd/pkg/types"
+)
+
+var (
+	linkStateMapMu sync.Mutex
+	linkStateMap   = make(map[linkName]linkState)
+)
+
+type linkName struct {
+	local, remote types.ID
+	src           string
+}
+
+type status bool
+
+const (
+	statusOK  status = true
+	statusBad status = false
+)
+
+type linkState struct {
+	status status
+	errm   map[string]bool
+}
+
+func reportLinkStatus(local, remote types.ID, src string, status status, err *opError) {
```

Comment:
moreover, if we already pass in the error why do we need the status field? error = nil implies statusOK?